src/app/lib/main.dart

```diff
@@ -29,6 +33,12 @@ import 'package:logger/logger.dart';
 late Logger _logger;
 
 Future<void> main() async {
+  await initializeComponents();
+
+  runApp(const App());
+}
+
+Future initializeComponents({bool? overrideIsMocked}) async {
```

Comment:
I think this should be renamed to `isMocked` to make it clearer.
- null keeps the default behavior
- true forces mock data
- false forces real data

Having the parameter named overrideXXX sounds a bit more confusing around what is the boolean about.